src/client.ts

```diff
@@ -1065,6 +1090,13 @@ export class AwaitConnection {
       this._abort();
     } finally {
       this._leaveOp();
+      this._cleanup();
```

Comment:
Is this the only way how the Connection object can be closed?  The method has to be explicitly called by the user, what if the Connection is closed because of a network error? Can we listen for some relevant event on `this.sock`?

Reply:
I registered an event listener to socket `onClose` event. I read your code several times in the existing `async close`, and it seems reasonable to call this method in the `onClose` event handler (I mean: not only the method that cleans the proxy), to ensure that all necessary operations are executed. Can you please confirm this?

Reply:
> Can you please confirm this?

Yeah, it's totally fine to call it from `close`, just need to make sure it's called consistently in all scenarios, both user-caused and due to an internal error. Listening on `onClose` should do the trick for both cases, no?

Reply:
I asked your confirmation because I was in doubt whether I should **only** do the proxy cleaning during the `onClose` socket event handler, rather than calling the `close` method that does more cleaning.